src/http/mod.rs

```diff
@@ -1403,6 +1433,55 @@ pub fn send_file<R: Read>(channel_id: u64, mut file: R, filename: &str, map: Jso
     serde_json::from_reader::<HyperResponse, Message>(response).map_err(From::from)
 }
 
+/// Sends file(s) to a channel.
+pub fn send_files<T: Into<FileOrPath>>(channel_id: u64, files: Vec<T>, map: JsonMap)
+    -> Result<Message> {
+    let uri = format!(api!("/channels/{}/messages"), channel_id);
+    let url = match Url::parse(&uri) {
+        Ok(url) => url,
+        Err(_) => return Err(Error::Url(uri)),
+    };
+
+    let mut request = Request::new(Method::Post, url)?;
+    request.headers_mut()
+        .set(header::Authorization(TOKEN.lock().unwrap().clone()));
+    request.headers_mut()
+        .set(header::UserAgent(constants::USER_AGENT.to_owned()));
+
+    let mut request = Multipart::from_request(request)?;
+    let mut file_num = String::from("0");
+    for file in files {
+        match file.into() {
+            FileOrPath::File(f) => {
+                let (mut f, filename) = f;
```

Comment:
We should instead move this up and do `FileOrPath::File((mut f, filename)) => {`